### PR TITLE
Hw 3

### PR DESCRIPTION
**Pull request by hellhellpnick**

### CODE REVIEW

File: homework-3/src/components/FormInput.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  name,
+  placeholder,
+  imgStart = null,
+  password = false,
+  err = false,
+  errText = '',
+  valueInput,
+  changer,
+}) => {
+  let [isType, setType] = React.useState(type);
```
Comment (codezavr) [inline homework-3/src/components/FormInput.js:19]: useState

File: homework-3/src/components/Forms/FormSignIn.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  return (
+    <Formik
+      initialValues={{
+        email: '',
+        password: '',
+      }}
+      validate={(values) => {
+        const errors = {};
+        const regulPassword = /(?=.*[A-Z]+)(?=.*[0-9])(?=.*[a-z]){6,}/;
+        const regulEmail =
```
Comment (codezavr) [inline homework-3/src/components/Forms/FormSignIn.js:24]: Move into constants and reuse

File: homework-3/src/components/Forms/FormSignIn.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+  return (
+    <Formik
+      initialValues={{
+        email: '',
+        password: '',
+      }}
+      validate={(values) => {
+        const errors = {};
+        const regulPassword = /(?=.*[A-Z]+)(?=.*[0-9])(?=.*[a-z]){6,}/;
```
Comment (codezavr) [inline homework-3/src/components/Forms/FormSignIn.js:23]: Move into constants and reuse

File: homework-3/src/components/Forms/FormSignIn.js [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+          setEmail((isEmail = true));
+        } else {
+          setEmail((isEmail = false));
+        }
+
+        return errors;
+      }}
+      onSubmit={(values, { setSubmitting }) => {
+        setTimeout(() => {
+          console.log(JSON.stringify(values, null, 2));
```
Comment (codezavr) [inline homework-3/src/components/Forms/FormSignIn.js:43]: REmove

File: homework-3/src/components/Forms/FormSignIn.js [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+        if (!regulEmail.test(values.email)) {
+          setEmail((isEmail = true));
+        } else {
+          setEmail((isEmail = false));
+        }
+
+        return errors;
+      }}
+      onSubmit={(values, { setSubmitting }) => {
+        setTimeout(() => {
```
Comment (codezavr) [inline homework-3/src/components/Forms/FormSignIn.js:42]: Something went wrong probably, setTimeout is the worst practice.

File: homework-3/src/components/Forms/FormSignUp.js [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+        firstName: '',
+        lastName: '',
+        email: '',
+        password: '',
+        checkPassword: '',
+      }}
+      validate={(values) => {
+        const errors = {};
+        const regulPassword = /(?=.*[A-Z]+)(?=.*[0-9])(?=.*[a-z]){6,}/;
+        const regulEmail =
```
Comment (codezavr) [inline homework-3/src/components/Forms/FormSignUp.js:28]: Move into constats and reuse

File: homework-3/src/components/Forms/FormSignUp.js [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+      initialValues={{
+        firstName: '',
+        lastName: '',
+        email: '',
+        password: '',
+        checkPassword: '',
+      }}
+      validate={(values) => {
+        const errors = {};
+        const regulPassword = /(?=.*[A-Z]+)(?=.*[0-9])(?=.*[a-z]){6,}/;
```
Comment (codezavr) [inline homework-3/src/components/Forms/FormSignUp.js:27]: Move into constats and reuse

File: homework-3/src/components/Forms/FormSignUp.js [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+          setEmail((isEmail = false));
+        }
+
+        return errors;
+      }}
+      onSubmit={(values, { setSubmitting }) => {
+        setTimeout(() => {
+          console.log(JSON.stringify(values, null, 2));
+          setSubmitting(false);
+        }, 400);
```
Comment (codezavr) [inline homework-3/src/components/Forms/FormSignUp.js:55]: What for do you use setTimeout?

File: homework-3/src/components/Forms/FormSignUp.js [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+          setEmail((isEmail = true));
+        } else {
+          setEmail((isEmail = false));
+        }
+
+        return errors;
+      }}
+      onSubmit={(values, { setSubmitting }) => {
+        setTimeout(() => {
+          console.log(JSON.stringify(values, null, 2));
```
Comment (codezavr) [inline homework-3/src/components/Forms/FormSignUp.js:53]: Remove

File: homework-3/src/views/DoctorProfile.js [hunk 10]
```diff
@@ -0,0 +1,242 @@
...
+import logo from '../img/logo.png';
+
+const DoctorProfile = ({ title }) => {
+  React.useEffect(() => {
+    document.title = title || '';
+  }, [title]);
+
+  const [loading, setLoading] = React.useState(true);
+
+  React.useEffect(() => {
```
Comment (codezavr) [inline homework-3/src/views/DoctorProfile.js:42]: Prevent importing React, just import useEffect.

File: homework-3/src/views/SignUp.js [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+
+  React.useEffect(() => {
+    const loadData = async () => {
+      await new Promise((r) => setTimeout(r, 2000));
+
+      setLoading((loading) => !loading);
+    };
+
+    loadData();
+  }, []);
```
Comment (codezavr) [inline homework-3/src/views/SignUp.js:23]: What is loading here didnt get it? what type of data is loading now?

**Review by codezavr — CHANGES_REQUESTED** [review]

File: homework-3/src/components/BtnsComponents/BtnEyePassword.js [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+  width: 20px;
+  height: 20px;
+  background-color: transparent;
+  border: none;
+  transition: all 0.2s ease;
+  z-index: 1;
+`;
+
+const BtnEyePassword = ({ func }) => {
+  const [isVisible, setVisible] = React.useState(true);
```
[the remark this answers is not in the document]
Comment (Lemout17) [reply homework-3/src/components/BtnsComponents/BtnEyePassword.js:21]: `import { useState } from 'react'`
`const [isVisible, setVisible] = useState(true);`

File: homework-3/src/components/BtnsComponents/BtnMore.js [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+  background-color: #dce0ec;
+  border-radius: 50%;
+`;
+
+const BtnMore = () => {
+  return (
+    <Btn type='button'>
+      <SpanDot />
+      <SpanDot />
+      <SpanDot />
```
[the remark this answers is not in the document]
Comment (Lemout17) [reply homework-3/src/components/BtnsComponents/BtnMore.js:32]: Just add svg as background-image for ur button

File: homework-3/src/views/SignIn.js [hunk 14]
```diff
@@ -0,0 +1,45 @@
...
+    const loadData = async () => {
+      await new Promise((r) => setTimeout(r, 2000));
+
+      setLoading((loading) => !loading);
+    };
+
+    loadData();
+  }, []);
+
+  if (loading) {
```
Comment (Lemout17) [inline homework-3/src/views/SignIn.js:25]: u can change if  block and set something like that 
`{loading ? <Loader /> :  all your components in else block }`
I think a simple ternary operator is better for understanding that if else block.

File: homework-3/src/views/PatientProfile.js [hunk 15]
```diff
@@ -0,0 +1,97 @@
...
+          </BoxBtnPatients>
+          <BoxRowPadding>
+            <TitlePatients>My Appointments</TitlePatients>
+            <BoxDoctor>
+              <BoxFilter text='Show: ' filter='Upcoming' />
+              <BtnCreateAppointment />
+            </BoxDoctor>
+          </BoxRowPadding>
+          <BoxPatientsList>
+            <CardPatient
```
Comment (Lemout17) [inline homework-3/src/views/PatientProfile.js:71]: code become clearer if u put all your proprs information about patient in db.json and u can leave only one props data={data}
in components CardPatient u can map this `data `props and create templates for your future cards
for Example :  `const CardPatient = ({data}) => {
   return (
      <ul>
           {data.map(({name,statusText,data,alt,img,info}) => {
               and here u create one template where u put all info from data
           })
      </ul>
   )
}`

File: homework-3/src/views/DoctorProfile.js [hunk 16]
```diff
@@ -0,0 +1,242 @@
...
+          </BoxBtnPatients>
+          <BoxRowPadding>
+            <TitlePatients>My patients</TitlePatients>
+            <BoxDoctor>
+              <BoxSearch />
+              <BoxFilter text='Sort by:' filter='Date' />
+            </BoxDoctor>
+          </BoxRowPadding>
+          <BoxPatientsList>
+            <CardPatient
```
Comment (Lemout17) [inline homework-3/src/views/DoctorProfile.js:80]: a lot of useless code here
create one component template and receive prop data where u can storage all info about the patient and map it 
ur code become clearer and readable

File: homework-3/src/views/PatientProfile.js [hunk 17]
```diff
@@ -0,0 +1,97 @@
...
+    };
+
+    loadData();
+  }, []);
+
+  if (loading) {
+    return <Loader />;
+  } else {
+    return (
+      <BoxPatients>
```
Comment (Lemout17) [inline homework-3/src/views/PatientProfile.js:46]: a lot of boilerplate code 
check [solution ](https://github.com/Lemout17/itrex-hw-3/blob/hw-3/src/components/mainDoctor/patientsList/patientsList.js) for better understanding

File: homework-3/src/Router.js [hunk 18]
```diff
@@ -0,0 +1,30 @@
...
+import { Route, Switch } from 'react-router-dom';
+import DoctorProfile from './views/DoctorProfile';
+import PatientProfile from './views/PatientProfile';
+import SignIn from './views/SignIn';
+import SignUp from './views/SignUp';
+import RestorePassword from './views/RestorePassword';
+
+const Router = () => (
+  <Switch>
+    <Route exact path='/'>
```
Comment (Lemout17) [inline homework-3/src/Router.js:11]: I think hardcoding path url is bad practice 
u can create routes.js file where u can store all your path strings in variables
Example: `export default {
  signupPage: '/',
  signinPage: '/signin',
  restorePage: '/restore',
  confirmPage: '/confirm',
  doctorPage: '/doctor',
  patientPage: '/patient',
}`
After that, u can import routes.js in Router.js component and change `Route exact path={routes.signupPage}`
If something changed in ur project, better to change the path in one variable.

File: homework-3/src/components/BtnsComponents/BtnMore.js (hunk 13 above)
[the remark this answers is not in the document]
Comment (matthew1232002) [reply homework-3/src/components/BtnsComponents/BtnMore.js:32]: It's better to create one more js file like BtnMore.styled.js and next to this file BtnMore.js . In BtnMore.js will be your component and BtnMore.styled.js will contain your styles.

File: homework-3/src/views/RestorePassword.js [hunk 19]
```diff
@@ -1,22 +1,39 @@
 import React from 'react';
 import { StylBoxMain } from '../components/common/Boxes/StylBoxMain';
-import FormRestorePassword from '../components/FormRestorePassword';
+import FormRestorePassword from '../components/Forms/FormRestorePassword';
 import BoxFormSignIn from './../components/common/Boxes/StylBoxFormSignIn';
+import Loader from './../components/Loader/Loader';
 
 const RestorePassword = ({ title }) => {
   React.useEffect(() => {
```
Comment (matthew1232002) [inline homework-3/src/views/RestorePassword.js:8]: You can import hooks directly

File: homework-3/src/views/RestorePassword.js [hunk 20]
```diff
@@ -1,22 +1,39 @@
...
-          <FormRestorePassword />
-        </BoxFormSignIn>
-      </StylBoxMain>
-    </>
-  );
+  const [loading, setLoading] = React.useState(true);
+
+  React.useEffect(() => {
+    const loadData = async () => {
+      await new Promise((r) => setTimeout(r, 2000));
```
Comment (matthew1232002) [inline homework-3/src/views/RestorePassword.js:16]: I heard that setTimeout is not a good practice

**Review by matthew1232002 — CHANGES_REQUESTED** [review]

File: homework-3/src/components/Boxes/BoxFilter.js [hunk 21]
```diff
@@ -0,0 +1,37 @@
...
+import arrowImg from './../../img/icons/arrow-up.png';
+import BtnFilter from './../common/Btns/StylBtnFilter';
+import FilterPatients from './../common/StylFilterPatients';
+import TextFilterPatient from './../common/Text/StylTextFilterPatient';
+import FilterTypePatient from './../common/StylFilterTypePatient';
+import ElementSelect from './../common/StylElementSelect';
+import BoxSelect from './../common/Boxes/StylBoxSelect';
+
+const BoxFilter = ({ text, filter }) => {
+  const [isHidden, setHidden] = React.useState(false);
```
[the remark this answers is not in the document]
Comment (hellhellpnick) [reply homework-3/src/components/Boxes/BoxFilter.js:13]: Import useState

File: homework-3/src/components/BtnsComponents/BtnEyePassword.js [hunk 22]
```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import styled from 'styled-components';
+import IconInputEnd from './../common/icons/StylIconInputEnd';
+import visibleEyeImg from './../../img/icons/icon-visible-password.svg';
+import closeEyeImg from './../../img/icons/icon-hidden-password.svg';
+
+const BtnEyePasswordStyle = styled.button`
```
[the remark this answers is not in the document]
Comment (hellhellpnick) [reply homework-3/src/components/BtnsComponents/BtnEyePassword.js:7]: Create new StylComponent `StylBtnEyePasswordStyle`

File: homework-3/src/components/BtnsComponents/BtnEyePassword.js (hunk 12 above)
[the remark this answers is not in the document]
Comment (hellhellpnick) [reply homework-3/src/components/BtnsComponents/BtnEyePassword.js:21]: Import useState

File: homework-3/src/components/BtnsComponents/BtnEyePassword.js [hunk 23]
```diff
@@ -0,0 +1,35 @@
...
+  transform: translateY(-50%);
+  width: 20px;
+  height: 20px;
+  background-color: transparent;
+  border: none;
+  transition: all 0.2s ease;
+  z-index: 1;
+`;
+
+const BtnEyePassword = ({ func }) => {
```
[the remark this answers is not in the document]
Comment (hellhellpnick) [reply homework-3/src/components/BtnsComponents/BtnEyePassword.js:20]: Rename props on `changeEye`

File: homework-3/src/components/FormInput.js (hunk 1 above)
Comment (hellhellpnick) [reply homework-3/src/components/FormInput.js:19]: Import useState

File: homework-3/src/components/Forms/FormSignIn.js (hunk 4 above)
Comment (hellhellpnick) [reply homework-3/src/components/Forms/FormSignIn.js:43]: Removed

File: homework-3/src/components/Forms/FormSignUp.js (hunk 8 above)
Comment (hellhellpnick) [reply homework-3/src/components/Forms/FormSignUp.js:55]: Removed setTimeout

File: homework-3/src/components/Forms/FormSignUp.js (hunk 9 above)
Comment (hellhellpnick) [reply homework-3/src/components/Forms/FormSignUp.js:53]: Removed

File: homework-3/src/views/DoctorProfile.js (hunk 10 above)
Comment (hellhellpnick) [reply homework-3/src/views/DoctorProfile.js:42]: Imported useEffect

File: homework-3/src/components/BtnsComponents/BtnMore.js (hunk 13 above)
[the remark this answers is not in the document]
Comment (hellhellpnick) [reply homework-3/src/components/BtnsComponents/BtnMore.js:32]: Fixed

File: homework-3/src/components/Forms/FormSignIn.js (hunk 5 above)
Comment (hellhellpnick) [reply homework-3/src/components/Forms/FormSignIn.js:42]: `setTimeout` deleted

File: homework-3/src/components/Forms/FormSignIn.js (hunk 3 above)
Comment (hellhellpnick) [reply homework-3/src/components/Forms/FormSignIn.js:23]: Moved

File: homework-3/src/components/Forms/FormSignIn.js (hunk 2 above)
Comment (hellhellpnick) [reply homework-3/src/components/Forms/FormSignIn.js:24]: Moved

File: homework-3/src/components/Forms/FormSignUp.js (hunk 6 above)
Comment (hellhellpnick) [reply homework-3/src/components/Forms/FormSignUp.js:28]: Moved

File: homework-3/src/components/Forms/FormSignUp.js (hunk 7 above)
Comment (hellhellpnick) [reply homework-3/src/components/Forms/FormSignUp.js:27]: Moved

File: homework-3/src/views/SignUp.js (hunk 11 above)
Comment (hellhellpnick) [reply homework-3/src/views/SignUp.js:23]: loadData was a promise, which is a kind of data. That's what he called it.
Now, useEffect with loader removed

File: homework-3/src/views/DoctorProfile.js (hunk 10 above)
Comment (hellhellpnick) [reply homework-3/src/views/DoctorProfile.js:42]: import useEffect

File: homework-3/src/views/SignIn.js (hunk 14 above)
Comment (hellhellpnick) [reply homework-3/src/views/SignIn.js:25]: Removed and moved to Router in component `Suspense`

File: homework-3/src/views/RestorePassword.js (hunk 19 above)
Comment (hellhellpnick) [reply homework-3/src/views/RestorePassword.js:8]: Imported

File: homework-3/src/components/Forms/FormRestorePassword.js [hunk 24]
```diff
@@ -0,0 +1,84 @@
...
+              Check your inbox, and click the reset link provided.
+            </InformationText>
+          </BoxViewContent>
+        </FormSign>
+      )}
+    </Formik>
+  );
+};
+
+export default FormRestorePassword;
```
[the remark this answers is not in the document]
Comment (hellhellpnick) [reply homework-3/src/components/Forms/FormRestorePassword.js:84]: Imported

File: homework-3/src/views/RestorePassword.js (hunk 20 above)
Comment (hellhellpnick) [reply homework-3/src/views/RestorePassword.js:16]: Removed

File: homework-3/src/views/PatientProfile.js (hunk 15 above)
Comment (hellhellpnick) [reply homework-3/src/views/PatientProfile.js:71]: Created 2 files `dbProfileDoctor` and `dbProfilePatient`

File: homework-3/src/views/DoctorProfile.js (hunk 16 above)
Comment (hellhellpnick) [reply homework-3/src/views/DoctorProfile.js:80]: Message above

File: homework-3/src/Router.js (hunk 18 above)
Comment (hellhellpnick) [reply homework-3/src/Router.js:11]: Added file `routes`

File: homework-3/src/components/CardPatientProfileDoctor/CardPatient.js [hunk 25]
```diff
@@ -1,32 +1,44 @@
...
-import TitleCard from './StylTitleCard';
-import SubTitleCard from './StylSubTitleCard';
 import BtnMore from './../BtnsComponents/BtnMore';
+
 import scheduleImg from './../../img/icons/schedule.png';
+import BoxPatientsList from './../../components/common/Boxes/StylBoxPatientsList';
 
-const CardPatient = ({ name, data, info, img, statusText, status, alt }) => {
+const CardPatient = ({ data }) => {
+  console.log(data[0]);
```
Comment (codezavr) [inline homework-3/src/components/CardPatientProfileDoctor/CardPatient.js:20]: Remove console.log

File: homework-3/src/components/CardPatientProfileDoctor/CardPatient.js [hunk 26]
```diff
@@ -1,32 +1,44 @@
...
-            <SubTitleCard status={status}>{statusText}</SubTitleCard>
-          </BoxColumnInfo>
-        </BoxDoctor>
-        <BtnMore />
-      </BoxRow>
-      <BoxDataPatient data={data} />
-      <BoxInfoPatient info={info} imgWay={scheduleImg} />
-    </BoxCardPatient>
+    <BoxPatientsList>
+      {data[0].map((item) => (
```
Comment (codezavr) [inline homework-3/src/components/CardPatientProfileDoctor/CardPatient.js:23]: data[0] it's your current patient, as i understood data[0] its the object, you should probably pass current Patient into this component, remove map and pass patient.

File: homework-3/src/components/CardPatientProfilePatient/CardProfilePatient.js [hunk 27]
```diff
@@ -1,32 +1,40 @@
...
-            <SubTitleCard>{statusText}</SubTitleCard>
-          </BoxColumnInfo>
-        </BoxDoctor>
-        <BtnMore />
-      </BoxRow>
-      <BoxDataPatient data={data} />
-      <BoxInfoPatient info={info} imgWay={heartImg} />
-    </BoxCardPatient>
+    <StylBoxPatientsList>
+      {data[0].map((item) => (
```
Comment (codezavr) [inline homework-3/src/components/CardPatientProfilePatient/CardProfilePatient.js:19]: Same as above pass current patient.

File: homework-3/src/components/Forms/FormRestorePassword.js [hunk 28]
```diff
@@ -1,19 +1,18 @@
...
 import emailImg from './../../img/icons/icon-email.svg';
-import LinkPage from '../common/StylLinkPage';
-import BoxViewContent from '../common/Boxes/StylBoxViewContent';
 
 const FormRestorePassword = () => {
-  let [isEmail, setEmail] = React.useState(false);
-  let [isHiddenForm, setHiddenForm] = React.useState(true);
+  const [isEmail, setEmail] = useState(false),
+    [isHiddenForm, setHiddenForm] = useState(true),
+    regulEmail =
```
Comment (codezavr) [inline homework-3/src/components/Forms/FormRestorePassword.js:14]: As i told you before, move this into constats.ts, and import into every component, as the constat.

File: homework-3/src/components/Forms/FormRestorePassword.js [hunk 29]
```diff
@@ -25,31 +24,25 @@ const FormRestorePassword = () => {
...
       validate={(values) => {
-        const regulEmail =
-          /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
-
         if (!regulEmail.test(values.email)) {
-          setEmail((isEmail = true));
+          setEmail(!isEmail);
         } else {
-          setEmail((isEmail = false));
+          setEmail(!isEmail);
```
Comment (codezavr) [inline homework-3/src/components/Forms/FormRestorePassword.js:30]: You di the same logic in if, probably will work wrong, test it and correct.

File: homework-3/src/components/Forms/FormSignIn.js [hunk 30]
```diff
@@ -1,16 +1,19 @@
...
 import passwordlImg from './../../img/icons/icon-lock.svg';
-import LinkPage from '../LinkPage';
 
 const FormSignUp = () => {
-  let [isEmail, setEmail] = React.useState(false);
-  let [isPassword, setPassword] = React.useState(false);
+  const [isEmail, setEmail] = useState(false),
+    [isPassword, setPassword] = useState(false),
+    regulPassword = /(?=.*[A-Z]+)(?=.*[0-9])(?=.*[a-z]){6,}/,
+    regulEmail =
```
Comment (codezavr) [inline homework-3/src/components/Forms/FormSignIn.js:15]: As i told you before, move this into constats.ts, and import into every component, as the constat.

File: homework-3/src/components/Forms/FormSignIn.js [hunk 31]
```diff
@@ -19,35 +22,25 @@ const FormSignUp = () => {
...
-        const regulPassword = /(?=.*[A-Z]+)(?=.*[0-9])(?=.*[a-z]){6,}/;
-        const regulEmail =
-          /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
-
         if (!regulPassword.test(values.password)) {
-          setPassword((isPassword = true));
+          setPassword(!isPassword);
         } else {
-          setPassword((isPassword = false));
+          setPassword(!isPassword);
```
Comment (codezavr) [inline homework-3/src/components/Forms/FormSignIn.js:28]: Logic does the same, recheck it will not work in a propper way

File: homework-3/src/components/Forms/FormSignIn.js [hunk 32]
```diff
@@ -19,35 +22,25 @@ const FormSignUp = () => {
...
         }
 
         if (!regulEmail.test(values.email)) {
-          setEmail((isEmail = true));
+          setEmail(!isEmail);
         } else {
-          setEmail((isEmail = false));
+          setEmail(!isEmail);
```
Comment (codezavr) [inline homework-3/src/components/Forms/FormSignIn.js:34]: Recheck

File: homework-3/src/components/Forms/FormSignUp.js [hunk 33]
```diff
@@ -1,17 +1,20 @@
...
 
 const FormSignUp = () => {
-  let [isEmail, setEmail] = React.useState(false);
-  let [isPassword, setPassword] = React.useState(false);
-  let [isCheckPassowrd, setCheckPassowrd] = React.useState(false);
+  const [isEmail, setEmail] = useState(false),
+    [isPassword, setPassword] = useState(false),
+    [isCheckPassowrd, setCheckPassowrd] = useState(false),
+    regulPassword = /(?=.*[A-Z]+)(?=.*[0-9])(?=.*[a-z]){6,}/,
+    regulEmail =
```
Comment (codezavr) [inline homework-3/src/components/Forms/FormSignUp.js:16]: constants.ts

File: homework-3/src/components/Forms/FormSignUp.js [hunk 34]
```diff
@@ -23,41 +26,31 @@ const FormSignUp = () => {
...
-        const regulPassword = /(?=.*[A-Z]+)(?=.*[0-9])(?=.*[a-z]){6,}/;
-        const regulEmail =
-          /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
-
         if (values.password !== values.checkPassword) {
-          setCheckPassowrd((isCheckPassowrd = true));
+          setCheckPassowrd(!isCheckPassowrd);
         } else {
-          setCheckPassowrd((isCheckPassowrd = false));
+          setCheckPassowrd(!isCheckPassowrd);
```
Comment (codezavr) [inline homework-3/src/components/Forms/FormSignUp.js:32]: Recheck duplicated logic

File: homework-3/src/theme/index.js [hunk 35]
```diff
@@ -6,4 +6,4 @@ const theme = {
   typography,
 };
 
-export default theme;
+export { theme };
```
Comment (codezavr) [inline homework-3/src/theme/index.js:9]: Why did you change export default?

**Review by codezavr — CHANGES_REQUESTED** [review]

File: homework-3/src/components/CardPatientProfileDoctor/CardPatient.js (hunk 25 above)
Comment (hellhellpnick) [reply homework-3/src/components/CardPatientProfileDoctor/CardPatient.js:20]: Removed

File: homework-3/src/components/CardPatientProfileDoctor/CardPatient.js (hunk 26 above)
Comment (hellhellpnick) [reply homework-3/src/components/CardPatientProfileDoctor/CardPatient.js:23]: No, data [0] its is an array of objects, so the map method is used

File: homework-3/src/components/CardPatientProfilePatient/CardProfilePatient.js (hunk 27 above)
Comment (hellhellpnick) [reply homework-3/src/components/CardPatientProfilePatient/CardProfilePatient.js:19]: Answer above

File: homework-3/src/components/Forms/FormRestorePassword.js (hunk 28 above)
Comment (hellhellpnick) [reply homework-3/src/components/Forms/FormRestorePassword.js:14]: oh, I misunderstood your idea
Fixed

File: homework-3/src/theme/index.js (hunk 35 above)
Comment (hellhellpnick) [reply homework-3/src/theme/index.js:9]: I don't remember why i changed this file
Fixed

File: homework-3/src/components/Forms/FormRestorePassword.js (hunk 29 above)
Comment (hellhellpnick) [reply homework-3/src/components/Forms/FormRestorePassword.js:30]: Added folder `constants` which has a file `index.js`

File: homework-3/src/components/Forms/FormSignIn.js (hunk 30 above)
Comment (hellhellpnick) [reply homework-3/src/components/Forms/FormSignIn.js:15]: Answer above

File: homework-3/src/components/Forms/FormSignIn.js (hunk 31 above)
Comment (hellhellpnick) [reply homework-3/src/components/Forms/FormSignIn.js:28]: Fixed

File: homework-3/src/components/Forms/FormSignIn.js (hunk 32 above)
Comment (hellhellpnick) [reply homework-3/src/components/Forms/FormSignIn.js:34]: Fixed

File: homework-3/src/components/Forms/FormSignUp.js (hunk 34 above)
Comment (hellhellpnick) [reply homework-3/src/components/Forms/FormSignUp.js:32]: Fixed

File: homework-3/src/components/Forms/FormSignUp.js (hunk 33 above)
Comment (hellhellpnick) [reply homework-3/src/components/Forms/FormSignUp.js:16]: Create file `index.js`
